### DISCUSSION TITLE
deepsource.io report, serverConfig is heavy (1056 bytes); consider passing it by pointer

### DISCUSSION THREAD

**Discussion by xiaods** [k3s-io/k3s#4990] [General]
https://github.com/k3s-io/k3s/blob/a094dee7dd0d7e7f7b2c8d50f90bb6760f9c86bf/pkg/cli/server/server.go#L495

![Image_20220119133255](https://user-images.githubusercontent.com/37678/150070107-dfe03039-73b6-41a3-9d6b-fd8a17155be9.png)

Is this a false alarm?

**Comment by brandond** [comment]
We do this once on startup, it's not exactly a performance critical operation. Changing it to take a pointer would be fine but I'm not sure it's worth it just for its own sake.

**Reply by xiaods** [reply]
I don't think this optimization is needed either.
